RUFAS/output_manager.py

```diff
@@ -893,6 +894,51 @@ def _route_save_functions(
                     info_map,
                 )
 
+    def _prepare_plot_data(self, filtered_pool: Dict[str, pool_element_type],
+                           selected_variables: Optional[List[str]] = None
+                           ) -> Dict[str, List[int | float]]:
+        """Extracts the values from the filtered pool data and converts them a dict format
+        that graph_generator can more readily handle.
+
+        Parameters
+        ----------
+        filtered_pool : Dict[str, pool_element_type]
+            The filtered pool of variables that the user wants to graph.
+        selected_variables : Optional[List[str]]
+            If it is present and the data is a list of dicts,
+            it selects the variables to be plotted.
+
+        Returns
+        -------
+        Dict[str, List[int | float]]
+            The formatted data that can more readily be plotted by graph_generator.
+        """
+        info_map = {
+            "class": self.__class__.__name__,
+            "function": self._prepare_plot_data.__name__,
+        }
+        prepared_pool = {}
+        for index, key in enumerate(filtered_pool.keys()):
+            values: List[Any] = filtered_pool[key]["values"]
+            is_data_in_dict = isinstance(values[0], dict)
+            if is_data_in_dict:
+                if selected_variables:
+                    data_dict = Utility.convert_list_of_dicts_to_dict_of_lists(values)
+                    for selected_variable in selected_variables:
+                        if selected_variable in data_dict:
+                            prepared_pool.setdefault(selected_variable, []).extend(data_dict[selected_variable])
+                else:
+                    self.add_error("No selected variables. Can't plot this data set.",
+                                   "List your 'variables' in the graph output filter.", info_map)
+                continue
```

Comment:
```suggestion
                if not selected_variables:
                    self.add_error("No selected variables. Can't plot this data set.",
                                   "List your 'variables' in the graph output filter.", info_map)
                    continue
                data_dict = Utility.convert_list_of_dicts_to_dict_of_lists(values)
                for selected_variable in selected_variables:
                    if selected_variable in data_dict:
                        prepared_pool.setdefault(selected_variable, []).extend(data_dict[selected_variable])
```
Not an important change, but reworking this branching statement to something similar to the above would make the code a little flatter.

Reply:
I can take a look. I think if the user hasn't added selected_variables to their filter file then we just want to break out of this loop (i.e. not continue) because we can't plot any of the data from that filter file (as of now). I think this will get addressed when I dive into issue #899.

Reply:
>Not an important change, but reworking this branching statement to something similar to the above would make the code a little flatter.

Checking for the unhappy path first is a good idea and one the best practices in writing code. It leads to a cleaner, more understandable, and maintainable code and is less prone to bugs.

Reply:
I've adjusted the code accordingly and put the unhappy path first (and just break the loop at that point since it's unnecessary to continue).